src/plugins/discourse/cacheKey.test.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+
+import {validateServerUrl} from "./models";
+import {cacheKey} from "./cacheKey";
+
+describe("plugins/discourse/cacheKey", () => {
+  describe("cacheKey", () => {
+    it("should work with valid server URLs", () => {
+      const corpus = [
+        "https://foo.bar",
+        "http://example.com/",
+        "HTTPS://Casing.WTF",
+        "https://custom.port.wtf:9001",
+        "https://discourse.sourcecred.io/",
+      ];
+      const results = corpus
+        .map(validateServerUrl)
+        .map((url) => [url, cacheKey(url)]);
+      expect(results).toMatchSnapshot();
```

Comment:
I would really prefer if we could avoid using snapshots for tests like
this. Snapshots have one advantage: when the test output is large, can
only reasonably be machine-generated, and is expected to be updated and
audited frequently, the snapshot can be helpful. But they have a bunch
of disadvantages. They move the expected output away from the input.
They move the expected output away from the code under test. They make
it harder to understand the repository history. They make it really easy
to write brittle tests, and they encourage cavalierly committing
snapshot updates.

I often read tests first. For instance, I cam to this test file to check
whether the outputs are expected to be lowercased. But after reading the
test file I still have no idea. Okay, I’ll check the snapshots—but there
are *two* snapshot files added in this change. Now I have to figure out
which one is which. All of this is extra friction. The expected output
should be right here with the inputs and test code.

Snapshots make it easy to write bad tests; resist the temptation. :-)

(You can take a look at #1685 for one clean way to do this.)

Reply:
Thanks, updated as request :]